### PR TITLE
3 remove empty constructor

### LINKED ISSUE #3: Remove empty constructors from Domain Model
There is a couple of empty constructors in the domain models, what is used to make Automapper happy.

We can remove that empty constructor following the guideline of [auto mapper](https://docs.automapper.org/en/stable/Custom-type-converters.html).

We also can think about removing Automapper. As we are not doing any complex mapping there.

### PR DESCRIPTION
**Pull request by ysnarafat**
Fix #3
- Remove Automapper reference
- Remove empty constructors
- Fix unit tests accordingly

### CODE REVIEW

File: InvoiceProcessor.Application/Queries/Outbox/GetPendingProcess/QueryHandler.cs [hunk 1]
```diff
@@ -12,19 +12,36 @@ public partial class GetPendingProcess
...
                 _outboxStorage = outboxStorage;
-                _mapper = mapper;
             }
 
             public async Task<List<Model>> Handle(Query request, CancellationToken cancellationToken)
             {
                 var pendingItems = await _outboxStorage.GetPendingItemsAsync(cancellationToken);
-                
-                return _mapper.Map<List<Model>>(pendingItems);
+                var result = PopulateData(pendingItems);
```
Comment (abuzaforfagun) [inline InvoiceProcessor.Application/Queries/Outbox/GetPendingProcess/QueryHandler.cs]: Insisted of doing the conversation manually, you can use Linq. The code would be more readable. And .net would select the best algorithm, which will help to improve the performance.
Ex: ```return pendingItems.Select(m => new Model(m.Guid.Value, m.CommandType, m.Data)).ToList();```

File: InvoiceReader.Application/InvoiceReader.Application.csproj [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="AutoMapper" Version="10.1.1" />
     <PackageReference Include="MediatR" Version="9.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Caching.Abstractions" Version="5.0.0" />
```
Comment (abuzaforfagun) [inline InvoiceReader.Application/InvoiceReader.Application.csproj]: What is the reason behind adding this reference?

File: InvoiceReader.Application/Queries/GetInvoices/Model.cs [hunk 3]
```diff
@@ -9,9 +10,34 @@ public record Result
...
+                            dueDate: invoiceItem.Attributes.DueDate,
+                            issueDate: invoiceItem.Attributes.Date,
+                            createdOn: invoiceItem.Attributes.DateCreated,
+                            updatedOn: invoiceItem.Attributes.DateUpdated
+                        );
+
+                        Data.Add(invoice);
+                    }
+                }
             }
```
Comment (abuzaforfagun) [inline InvoiceReader.Application/Queries/GetInvoices/Model.cs]: Our record should not deal with InvoiceItem.
It should accept ```List<Invoice>``` as an argument. And assign that accordingly.

**Review by abuzaforfagun — CHANGES_REQUESTED** [review]
I have added some code review feedbacks.

Can you please fix those?

Thanks

File: InvoiceReader.Application/InvoiceReader.Application.csproj (hunk 2 above)
Comment (ysnarafat) [reply InvoiceReader.Application/InvoiceReader.Application.csproj]: I had problem in building code without this package. I tried without installing it but didn't worked, that's why added.

**Comment by ysnarafat** [comment]
Yes, I will do accordingly. Thanks for the feedback.

File: InvoiceReader.Application/InvoiceReader.Application.csproj (hunk 2 above)
Comment (abuzaforfagun) [reply InvoiceReader.Application/InvoiceReader.Application.csproj]: Okay, I understand the issue.

It has a dll reference. And this is the reason why it's working on my machine and not yours.
```
<ItemGroup>
    <Reference Include="Microsoft.Extensions.Caching.Abstractions">
      <HintPath>..\..\..\..\..\..\Program Files\dotnet\packs\Microsoft.AspNetCore.App.Ref\5.0.0\ref\net5.0\Microsoft.Extensions.Caching.Abstractions.dll</HintPath>
    </Reference>
  </ItemGroup>
  ```
Can you please remove your changes, I will create a separate issue regarding this. Fix that accordingly.

Comment (ysnarafat) [reply InvoiceReader.Application/InvoiceReader.Application.csproj]: Removed reference

File: InvoiceReader.Application/Queries/GetInvoices/Model.cs [hunk 4]
```diff
@@ -9,9 +10,9 @@ public record Result
         {
             public List<Invoice> Data { get; set; }
```
Comment (abuzaforfagun) [inline InvoiceReader.Application/Queries/GetInvoices/Model.cs]: One more thing, can you please use init set, insisted of the public set.
Public set could break the DDD principles for our model.

Comment (ysnarafat) [reply InvoiceReader.Application/Queries/GetInvoices/Model.cs]: Yes, I will.

**Review by abuzaforfagun — APPROVED** [review]

**Pull request merged**